code/datums/uplink_items/uplink_general.dm

```diff
@@ -22,25 +20,49 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 				uplink_items[I.category] = list()
 
 			uplink_items[I.category] += I
-			if(I.limited_stock < 0 && I.can_discount && I.item && I.cost > 5)
+
+			//Add items to discount pool, checking job, species, and hijacker status
+			var/job_check = FALSE
+			var/species_check = FALSE
+			var/hijacker_check = FALSE
+			if(I.job) //If your job does not match, no discount
+				if(user.mind.assigned_role in I.job)
+					job_check = TRUE
+
+			else
+				job_check = TRUE
+			if(I.species) //If your species does not match, no discount
+				if(user.dna?.species.name in I.species)
+					species_check = TRUE
+			else
+				species_check = TRUE
+			if(I.hijack_only && !(locate(/datum/objective/hijack) in user.mind.get_all_objectives())) //If you aren't a hijacker, no hijack only items
+				hijacker_check = TRUE
+			else if(!I.hijack_only)
+				hijacker_check = TRUE
```

Comment:
I think it might be better to do this in a style like this
```js
if(I.job && !(user.mind.assigned_role in I.job))
	continue
```
This way we still don't add wrong items to the potential discounts but we don't do a massive check in the end

Reply:
Made this change, and moved the checks above the category assignment.